fivox/eventSource.cpp

```diff
@@ -125,26 +218,26 @@ Events EventSource::findEvents( const AABBf& area LB_UNUSED ) const
         _impl->rtree.query( bgi::intersects( query ), std::back_inserter( hits ));
         maxHits = std::max( size_t(maxHits), hits.size( ));
 
-        Events events;
-        events.reserve( hits.size( ));
+        eventValues.reserve( hits.size( ));
         for( const Value& value : hits )
-        {
-            const Event& val = _impl->events[ value.second ];
-            if( val.value != VALUE_UNSET )
-                events.push_back( val );
-        }
-        return events;
+            eventValues.push_back( getValues()[value.second] );
     }
+    else
 #endif
-
-    static bool first = true;
-    if( first )
+    // return all values
```

Comment:
actually, in the functors where we use findEvents(), returning all event values is giving a completely different result if the functor is not 'clipping' events that are outside the requested region. I would suggest to exclude those functors if no rtree is available or implement a slow event filtering here.